### ISSUE TITLE
Gradient do not exists for variables ['offset_reg/kernel:0', 'offset_reg/bias:0'] when minimizing the loss

### ISSUE THREAD

**Issue by lyiqian** [liu1000/pointnetpp-tensorflow2#1]
When training, `output_2_loss` would actually increase.

```
Loading data..
Loading previously saved TF Datasets
Building..
Model: "pienet_pt_model"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
=================================================================
 pointnet_sa (Pointnet_SA)   multiple                  3680      
                                                                 
 pointnet_sa_1 (Pointnet_SA)  multiple                 17600     
                                                                 
 pointnet_sa_2 (Pointnet_SA)  multiple                 67968     
                                                                 
 pointnet_sa_3 (Pointnet_SA)  multiple                 267008    
                                                                 
 pointnet_fp (Pointnet_FP)   multiple                  264192    
                                                                 
 pointnet_fp_1 (Pointnet_FP)  multiple                 165888    
                                                                 
 pointnet_fp_2 (Pointnet_FP)  multiple                 116224    
                                                                 
 pointnet_fp_3 (Pointnet_FP)  multiple                 50688     
                                                                 
 dense (Dense)               multiple                  16512     
                                                                 
 dropout (Dropout)           multiple                  0         
                                                                 
 point_clf (Dense)           multiple                  258       
                                                                 
 offset_reg (Dense)          multiple                  387       
                                                                 
=================================================================
Total params: 970,405
Trainable params: 962,981
Non-trainable params: 7,424
_________________________________________________________________
None
Compiling..
Fitting..
Epoch 1/100
WARNING:tensorflow:Gradients do not exist for variables ['offset_reg/kernel:0', 'offset_reg/bias:0'] when minimizing the loss. If you're using `model.compile()`, did you forget to provide a `loss` argument?
WARNING:tensorflow:Gradients do not exist for variables ['offset_reg/kernel:0', 'offset_reg/bias:0'] when minimizing the loss. If you're using `model.compile()`, did you forget to provide a `loss` argument?
WARNING:tensorflow:Gradients do not exist for variables ['offset_reg/kernel:0', 'offset_reg/bias:0'] when minimizing the loss. If you're using `model.compile()`, did you forget to provide a `loss` argument?
WARNING:tensorflow:Gradients do not exist for variables ['offset_reg/kernel:0', 'offset_reg/bias:0'] when minimizing the loss. If you're using `model.compile()`, did you forget to provide a `loss` argument?
   1/1254 [..............................] - ETA: 2:40:23 - loss: 0.4282 - output_1_loss: 0.1585 - output_2_loss: 0.2697 - output_1_sparse_categorical_accuracy: 0.95
   2/1254 [..............................] - ETA: 1:52 - loss: 2.7544 - output_1_loss: 0.1190 - output_2_loss: 2.6354 - output_1_sparse_categorical_accuracy: 0.9739
   3/1254 [..............................] - ETA: 1:40 - loss: 3.0014 - output_1_loss: 0.1007 - output_2_loss: 2.9008 - output_1_sparse_categorical_accuracy: 0.9783
   4/1254 [..............................] - ETA: 1:42 - loss: 2.9217 - output_1_loss: 0.0812 - output_2_loss: 2.8405 - output_1_sparse_categorical_accuracy: 0.9817
   5/1254 [..............................] - ETA: 1:38 - loss: 2.8214 - output_1_loss: 0.0908 - output_2_loss: 2.7306 - output_1_sparse_categorical_accuracy: 0.9839 
...
```
